WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPersonalizationService.swift

```diff
@@ -15,6 +28,47 @@ struct BlogDashboardPersonalizationService {
         self.siteID = String(siteID)
     }
 
+    // MARK: - Core API
+
+    func setEnabled(_ isEnabled: Bool, forKey key: String, scope: SettingsScope) {
+        var settings = getSettings(for: key)
+        let lookUpKey = lookUpKey(from: scope)
+        settings[lookUpKey] = isEnabled
+        self.repository.set(settings, forKey: key)
+        NotificationCenter.default.post(name: .blogDashboardPersonalizationSettingsChanged, object: self)
+    }
```

Comment:
The logic it self didn't change. See [previous implementation](https://github.com/wordpress-mobile/WordPress-iOS/pull/22229/files#diff-6f57c9c50e41d9cea47e043fc0978eb3d2c4300aeabfd793b0ef92b358436950L59).

What has changed, however, is the API. The prior method's API was closely tied to objects of the DashboardCard type, which limited its applicability to cards of different types, such as DashboardDynamicCardModel.

To address this, the API has been redesigned to be more abstract, allowing it not to be confined to a specific type.